Tweak.x

```diff
@@ -415,8 +416,38 @@ NSDictionary *getIdentifiers() {
 
 	// This %ctor will be called every time identityservicesd or the settings app is restarted.
 	// So we only want it to try to reinitialize stuff if it's in identityservicesd
-	if (![bundleID isEqualToString:@"com.apple.identityservicesd"])
+	if (![bundleID isEqualToString:@"com.apple.identityservicesd"]) {
+		[[NSDistributedNotificationCenter defaultCenter] addObserverForName: @"com.beeper.beepserv/updateState"
+			object: nil
+			queue: [NSOperationQueue mainQueue]
+			usingBlock: ^(NSNotification *notification)
+		{
+			NSDictionary *state = notification.userInfo;
+			LOG(@"Received broadcasted state: %@", state);
+			currentState = [BPState.alloc initWithCode:[state valueForKey: @"com.beepserv.code"] secret:[state valueForKey: @"com.beepserv.secret"] connected:[state valueForKey: @"com.beepserv.connected"] error:[state valueForKey: @"com.beepserv.error"]];
+			NSError *diskWriteError;
+			[currentState writeToDiskWithError:&diskWriteError];
+			if (diskWriteError != nil) {
+				LOG(@"Writing state to disk failed with error: %@", diskWriteError);
+			}
+		}];
+		[[NSDistributedNotificationCenter defaultCenter]
+			postNotificationName: @"com.beeper.beepserv/requestStateUpdate"
+			object: nil
+			userInfo: nil
+		];
 		return;
+	}
+	
+	[[NSDistributedNotificationCenter defaultCenter] addObserverForName: @"com.beeper.beepserv/requestStateUpdate"
```

Comment:
is `defaultCenter` guaranteed to never be de-allocated? And do you know if there's a possibility that these listener blocks might be deallocated, causing us to run into an issue where Settings and identityservicesd can't communicate with each other?

Reply:
From previous experiences using `[NSDistributedNotificationCenter defaultCenter]` in tweaks, I've never ran into a problem with it or the block getting de-allocated, so as far as I know this should not be an issue, but you're right, if that were to happen, we would run into a problem where the two can't communicate until `identityservicesd` is restarted.

Reply:
Maybe we should fall back to trying to read the state file from disk when the Settings app does not receive a status from `identityservicesd`?

Reply:
we could... if you've already tested this new system out and verified that it fixes your previous problem with not having permissions (or rather, works around that problem), then i don't think we need to do that and can just keep it in mind in case we start seeing issues where Settings doesn't have the code.

Reply:
Yeah, with these changes I was able to register Beeper Mini with my iPad. If we start seeing such issues, we could always add that fallback later.